doc/xbgpu.design.b.rst

```diff
@@ -86,3 +86,35 @@ scaling to convert to a real value in :math:`(-0.5, 0.5)`. While this is
 still a deviation from uniformity, it does give a symmetric distribution.
 
 .. _curand: https://docs.nvidia.com/cuda/curand/index.html
+
+Data flow
+---------
+The host side of the beamforming is simpler than for correlation, because
+there is no accumulation. For simplicity, the output chunk size (in time) is
+set to the same as the input chunk size.
+
+Because the kernel operates on all beams together, there is only one instead
+of the :class:`.BPipeline` class, and it handles all the beams. This is in
+contrast to :class:`.XPipeline`, which only handles a single :class:`Output`.
+
+Unlike the correlation pipeline, the beamformer pipeline can be controlled
+dynamically by setting weights and gains. Transferring these to the GPU is
+somewhat expensive and should not be done for every chunk. Instead, there is
+an associated version number. When the weights/gains are updated, the master
+version number is incremented. Before processing a chunk on the GPU, this
+version number is compared to a version number associated with the GPU copy,
+and if they differ then an update is performed.
```

Comment:
Excellent summation and barely a complaint about this whole paragraph. If I'm really nit-picking, the final third of the final sentence could be decoupled. May I suggest
> Before processing a chunk on the GPU, [...] associated with the GPU copy. If they differ, an update is performed.

It generally follows the flow of the second ~half of the paragraph
* Reason-comma-action.